human-dynamics-estimator/src/HumanDynamicsEstimator.cpp

```diff
@@ -371,23 +376,26 @@ bool HumanDynamicsEstimator::updateModule()
  
     sigma_BarD_inv = (toEigen(m_D).transpose() * toEigen(m_sigmaD_prior) * toEigen(m_D)) + toEigen(m_sigmad_prior); 
    
-    if(m_count==1) {                    
-    m_llt.analyzePattern(sigma_BarD_inv);  
+    if(m_firststep == true) 
+    {                    
+        m_CholeskyLU_sigma_BarD_inv.analyzePattern(sigma_BarD_inv);  
     }
-    m_llt.factorize(sigma_BarD_inv); 
+    m_CholeskyLU_sigma_BarD_inv.factorize(sigma_BarD_inv); 
     
     rhs_BarD = (toEigen(m_sigmad_prior) * toEigen(m_mud_prior)) - ((toEigen(m_D).transpose() * (toEigen(m_sigmaD_prior) * toEigen(m_bD))));
-    mu_BarD = m_llt.solve(rhs_BarD); 
+    mu_BarD = m_CholeskyLU_sigma_BarD_inv.solve(rhs_BarD); 
     
     sigma_dgiveny_inv = sigma_BarD_inv + (toEigen(m_Y).transpose() * (toEigen(m_sigmay_prior) * toEigen(m_Y)));
       
-    if(m_count==1) {
-    m_R.analyzePattern(sigma_dgiveny_inv);   
+    if(m_firststep==true) 
```

Comment:
I think it is better to collect all the code that needs to be run once in the same point.
So move these lines to the `if` at line 379